src/routes/webhooks/paddle.ts

```diff
@@ -74,6 +82,39 @@ const updateUserSubscription = async ({
     );
     return false;
   }
+
+  const { gifter_id: gifterId, duration } = customData;
+  const durationTime = duration && parseInt(duration);
+  const isGift = !!durationTime && gifterId;
+  if (isGift) {
+    if (userId === gifterId) {
+      logger.error({ type: 'paddle', data }, 'User and gifter are the same');
+      return false;
+    }
+
+    if (!durationTime || durationTime <= 0) {
+      logger.error({ type: 'paddle', data }, 'Invalid duration');
+      return false;
+    }
+
+    const gifterUser = await con
+      .getRepository(User)
+      .findOneBy({ id: gifterId });
+    if (!gifterUser) {
+      logger.error({ type: 'paddle', data }, 'Gifter user not found');
+      return false;
+    }
+
+    const targetUser = await con.getRepository(User).findOne({
+      select: ['subscriptionFlags'],
+      where: { id: userId },
+    });
+    if (isPlusMember(targetUser?.subscriptionFlags?.cycle)) {
+      logger.error({ type: 'paddle', data }, 'User is already a Plus member');
+      return false;
+    }
+  }
```

Comment:
This means that even though user is already member the gifter payed so not sure if we need to do anything there money wise?

Reply:
User shouldn't be able to gift to a plus user, this is a really edge case imo 👀

Reply:
I agree with Ante's concern. Should we maybe record this somewhere? Or the logging would enough? Maybe we need to send a Slack message or something.